panic: runtime error: invalid memory address or nil pointer dereference

Hi, thank you for the fantastic tool. I love using git-spice and constantly praise it to my teammates.

I'm getting a panic when running a basic `gs ll`, like so –

```
❯ gs log long
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x1028c0070]

goroutine 51 [running]:
go.abhg.dev/gs/internal/spice.(*Service).LoadBranches.func1()
	go.abhg.dev/gs/internal/spice/branch.go:453 +0x330
sync.(*WaitGroup).Go.func1()
	sync/waitgroup.go:239 +0x4c
created by sync.(*WaitGroup).Go in goroutine 1
	sync/waitgroup.go:237 +0x70
```

The same error occurs for commands:
- `gs log long`
- `gs bco` with no arguments
- `gs up`
- `gs br`

but `gs version` and `gs auth status` work without issue. In a different repo, everything works fine – which makes me think I need to "reset" git-spice's internal state for this repo, if such a thing exists, but as you can probably tell I'm really just guessing. I'd appreciate any help you can provide!

edit: if it helps, pushing my branch locally and opening a PR (in a private repo) shows this warning at the top in Github. I've never seen this before and wonder if this has to do with the branch name!

```
The head ref may contain hidden characters: "tushar/domain\u2013payables-egress"
```

--- a/.changes/unreleased/Fixed-20251103-162308.yaml
+++ b/.changes/unreleased/Fixed-20251103-162308.yaml
@@ -1,0 +1,3 @@
+kind: Fixed
+body: Fix crash when information for a branch could not be loaded from Git
+time: 2025-11-03T16:23:08.7112-08:00

--- a/internal/spice/branch.go
+++ b/internal/spice/branch.go
@@ -443,6 +443,7 @@ func (s *Service) LoadBranches(ctx context.Context) ([]LoadBranchItem, error) {
 					mu.Lock()
 					errs = append(errs, fmt.Errorf("get branch %v: %w", name, err))
 					mu.Unlock()
+					continue
 				}
 
 				mu.Lock()

--- a/internal/spice/branch_test.go
+++ b/internal/spice/branch_test.go
@@ -274,3 +274,37 @@ func TestService_LookupBranch_upstreamBranch(t *testing.T) {
 		assert.Empty(t, lookup.UpstreamBranch)
 	})
 }
+
+// Regression test for https://github.com/abhinav/git-spice/issues/926
+func TestService_LoadBranches_lookupError_issue926(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+
+	mockStore := NewMockStore(mockCtrl)
+
+	mockStore.EXPECT().
+		ListBranches(gomock.Any()).
+		Return(func(yield func(string, error) bool) {
+			yield("feature", nil)
+		})
+
+	mockStore.EXPECT().
+		LookupBranch(gomock.Any(), "feature").
+		Return(nil, assert.AnError)
+
+	mockRepo := NewMockGitRepository(mockCtrl)
+	mockRepo.EXPECT().
+		PeelToCommit(gomock.Any(), "feature").
+		Return(git.Hash("abc123"), nil)
+
+	svc := NewService(
+		mockRepo,
+		NewMockGitWorktree(mockCtrl),
+		mockStore,
+		nil,
+		silogtest.New(t),
+	)
+	ctx := t.Context()
+	assert.NotPanics(t, func() {
+		_, _ = svc.LoadBranches(ctx)
+	})
+}

